cmake/urdfdom-config.cmake.in

```diff
@@ -7,16 +7,31 @@ set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include" "@TinyXML_INCLUDE_D
 
 foreach(lib @PKG_LIBRARIES@)
   set(onelib "${lib}-NOTFOUND")
+  set(onelibd "${lib}-NOTFOUND")
   find_library(onelib ${lib}
       PATHS "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@"
     NO_DEFAULT_PATH
     )
-  if(NOT onelib)
+  find_library(onelibd ${lib}@CMAKE_DEBUG_POSTFIX@
+      PATHS "@CMAKE_INSTALL_PREFIX@/@CMAKE_INSTALL_LIBDIR@"
+    NO_DEFAULT_PATH
+    )
+  if(onelib-NOTFOUND AND onelibd-NOTFOUND)
     message(FATAL_ERROR "Library '${lib}' in package @PKG_NAME@ is not installed properly")
   endif()
-  list(APPEND @PKG_NAME@_LIBRARIES ${onelib})
+  if (onelib)
+    list(APPEND @PKG_NAME@_RELEASE_LIBRARIES ${onelib})
+  endif()
+  if (onelibd)
+    list(APPEND @PKG_NAME@_DEBUG_LIBRARIES ${onelibd})
+  endif()
 endforeach()
-
+if(@PKG_NAME@_RELEASE_LIBRARIES)
+  list(APPEND @PKG_NAME@_LIBRARIES optimized ${@PKG_NAME@_RELEASE_LIBRARIES})
```

Comment:
The build configuration keyword must be stated before each library name (not only once in front of multiple names).

I would suggest to fill `@PKG_NAME@_LIBRARIES` directly in line 23 and 26 and avoid the temporary variables `@PKG_NAME@_RELEASE_LIBRARIES` and `@PKG_NAME@_DEBUG_LIBRARIES`.